client/engine/store/memstore.go

```diff
@@ -179,6 +179,25 @@ func (ms *MemStore) getChannelById(id types.Destination) (channel.Channel, error
 	return ch, nil
 }
 
+// GetChannelsByAppDefinition returns any channels that include the given participant
+func (ms *MemStore) GetChannelsByAppDefinition(appDef types.Address) []*channel.Channel {
+	toReturn := []*channel.Channel{}
+	ms.channels.Range(func(key string, chJSON []byte) bool {
+		var ch channel.Channel
+		err := json.Unmarshal(chJSON, &ch)
+		if err != nil {
+			return true // channel not found, continue looking
```

Comment:
There's different behaviour here in the memstore compared to the durablestore. In the latter, an unmarshaling error caused a panic. Here, it is essentially swallowed and we carry on.

Reply:
I think the behaviour here is incorrect, it should panic/error like the memstore.